plugin-infra/go-plugin-infra/src/main/java/com/thoughtworks/go/plugin/infra/DefaultPluginManager.java

```diff
@@ -85,6 +85,11 @@ public PluginUploadResponse addPlugin(File uploadedPlugin, String filename) {
         return pluginWriter.addPlugin(uploadedPlugin, filename);
     }
 
+    @Override
+    public boolean isPluginLoaded(String pluginId) {
+        return getPluginDescriptorFor(pluginId) != null;
```

Comment:
This is not good enough, since there could be plugins installed but not loaded. Add check for `getPluginDescriptorFor(pluginId).isInvalid()` too, and ofcourse a test for the same

Reply:
Make sense to me. I will change it.